### PR TITLE
add basic support files for rtl87x2g series soc

### PR DESCRIPTION
**Pull request by YuzhuoLiuRTK**
In this PR, I add code for supporting the realtek rtl87x2g series soc.
With this PR, you can build and run a hello_world app with rtl8762gn_evb.
You can also use drivers such as urat, dma, gpio, adc and counter with rtl87x2g series soc.
Hi @RSGraceGuan @ZhiyuanTang17 , Please help to review this PR.

### CODE REVIEW

File: boards/arm/rtl8762gn_evb/rtl8762gn_evb.dts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/dts-v1/;
+#include <rtk/rtl8762gn_evb.dtsi>
```
Comment (ZhiyuanTang17) [inline boards/arm/rtl8762gn_evb/rtl8762gn_evb.dts]: Please change the dts/arm/rtk/rtl8762gn_evb.dtsi to dts/arm/**realtek**/rtl8762gn_evb.dtsi

File: boards/arm/rtl8762gn_evb/rtl8762gn_evb.dts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#include "rtl8762gn_evb-pinctrl.dtsi"
+
+/ {
+	model = "RealTek rtl8762gn evb ";
+	compatible = "rtk,rtl8762gn_evb";
```
Comment (ZhiyuanTang17) [inline boards/arm/rtl8762gn_evb/rtl8762gn_evb.dts]: Please change the "rtk,rtl8762gn_evb" to"realtek,rtl8762gn_evb"

**Review by ZhiyuanTang17 — COMMENTED** [review]
I test hello_world app with this PR https://github.com/rtkconnectivity/hal_realtek/pull/4 
And hello_world app works well.
There are some places need to change rtk to realtek，please help to modify.

File: boards/arm/rtl8762gn_evb/rtl8762gn_evb.dts (hunk 2 above)
Comment (YuzhuoLiuRTK) [reply boards/arm/rtl8762gn_evb/rtl8762gn_evb.dts]: Done, please help review

File: boards/arm/rtl8762gn_evb/rtl8762gn_evb.dts (hunk 1 above)
Comment (YuzhuoLiuRTK) [reply boards/arm/rtl8762gn_evb/rtl8762gn_evb.dts]: Done

**Review by ZhiyuanTang17 — APPROVED** [review]
I am ok with this PR merged. Please let @RSGraceGuan help to review.

**Pull request closed without merging**